core/base/v7/test/base_exception.cxx

```diff
@@ -0,0 +1,105 @@
+#include "gtest/gtest.h"
+
+#include <ROOT/RError.hxx>
+
+#include <cerrno>
+#include <stdexcept>
+
+using RException = ROOT::Experimental::RException;
+using RStatusBool = ROOT::Experimental::RStatusBool;
+using RStatusSyscall = ROOT::Experimental::RStatusSyscall;
+
+namespace {
+
+static RStatusBool TestFailure()
+{
+   return RStatusBool(false);
+}
+
+static RStatusBool TestSuccess()
+{
+   return RStatusBool(true);
+}
+
+static RStatusSyscall MockFileOpen(bool succeed)
+{
+   if (succeed)
+      return RStatusSyscall(42);
+   errno = EPERM;
+   return RStatusSyscall::Fail("Not allowed to succeed");
+}
+
+class ExceptionX : public std::runtime_error {
+public:
+   explicit ExceptionX(const std::string &what) : std::runtime_error(what) {}
+};
+
+} // anonymous namespace
+
+
+TEST(Exception, InstantExceptions)
+{
+   ROOT::Experimental::SetThrowInstantExceptions(true); // the default
+   bool passedFailure = false;
+
+   try {
+      TestFailure();
+      passedFailure = true;
+   } catch (const RException&) {
+   }
```

Comment:
For 'fun' shouldn't we also add an assignment in the catch close to check it is run?